src/Features/CSharp/Portable/Completion/CompletionProviders/InternalsVisibleToCompletionProvider.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.Completion.Providers;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Microsoft.CodeAnalysis.CSharp.Completion.Providers
+{
+    internal sealed class InternalsVisibleToCompletionProvider : AbstractInternalsVisibleToCompletionProvider
+    {
+
+        protected override IImmutableList<SyntaxNode> GetAssemblyScopedAttributeSyntaxNodesOfDocument(SyntaxNode documentRoot)
+        {
+            var builder = default(ImmutableList<SyntaxNode>.Builder);
+            if (documentRoot is CompilationUnitSyntax compilationUnit)
+            {
+                foreach (var attributeList in compilationUnit.AttributeLists)
+                {
+                    // For most documents the compilationUnit.AttributeLists should be empty.
+                    // Therefore we delay initialization of the builder
+                    builder = builder ?? ImmutableList.CreateBuilder<SyntaxNode>();
+                    builder.AddRange(attributeList.Attributes);
+                }
+            }
+
+            return builder == null
+                ? ImmutableList<SyntaxNode>.Empty
+                : builder.ToImmutable();
+        }
+
+        protected override SyntaxNode GetConstructorArgumentOfInternalsVisibleToAttribute(SyntaxNode internalsVisibleToAttribute)
+        {
+            if (internalsVisibleToAttribute is AttributeSyntax attributeSyntax)
+            {
+                // InternalsVisibleTo has only one constructor argument. 
+                // https://msdn.microsoft.com/en-us/library/system.runtime.compilerservices.internalsvisibletoattribute.internalsvisibletoattribute(v=vs.110).aspx
+                // We can assume that this is the assemblyName argument.
+                foreach (var argument in attributeSyntax.ArgumentList.Arguments)
```

Comment:
To me the loop makes it look like you have to search the argument list to find the one that contains the assembly name, but it's actually always going to be the first one.